pkg/controller/pkg/revision/reconciler.go

```diff
@@ -153,8 +152,8 @@ func BuildObjectScheme() (*runtime.Scheme, error) {
 
 // Reconciler reconciles packages.
 type Reconciler struct {
-	namespace    string
-	client       resource.ClientApplicator
+	client       client.Client
+	hook         Hook
```

Comment:
Nit: Maybe `hook Hooks` since the `Hooks` type actually has two hooks (pre and post), but when you call them it would be `r.hook.Pre()`, which reads like "run the pre hook".